File: 940-fruit-into-baskets/fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        d = dict()
        start = 0
        max_count = 0
        for end in range(len(fruits)):
            fruit = fruits[end]
            d[fruit] = d.get(fruit, 0) + 1

            while(len(d)>2):
                d[fruits[start]] -= 1
                if(d[fruits[start]] == 0):
                    d.pop(fruits[start])
                start += 1
            
            max_count = max(max_count, end-start+1)
        
        return max_count

File: 940-fruit-into-baskets/fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        unordered_map<int, int> mp;
        int start = 0;
        int max_count = 0;

        for(int end = 0; end < fruits.size(); end++){
            mp[fruits[end]]++;
            while (mp.size() > 2) {
                mp[fruits[start]]--;
                if (mp[fruits[start]] == 0)
                    mp.erase(fruits[start]);
                start++;
            }

            max_count = max(max_count, end-start+1);
        }

        return max_count;
    }
};